### PR TITLE
ddt removal in ui discovery rule tests

### PR DESCRIPTION
**Pull request by sthirugn**
Test results:
```sh
# nosetests tests/foreman/ui/test_discoveryrule.py:DiscoveryRules.test_positive_create_discovery_rule_1
.
----------------------------------------------------------------------
Ran 1 test in 31.983s

OK

# nosetests tests/foreman/ui/test_discoveryrule.py:DiscoveryRules.test_negative_create_discovery_rule_2
.
----------------------------------------------------------------------
Ran 1 test in 39.206s

OK

# nosetests tests/foreman/ui/test_discoveryrule.py:DiscoveryRules.test_negative_create_discovery_rule_3
.
----------------------------------------------------------------------
Ran 1 test in 39.648s

OK

# nosetests tests/foreman/ui/test_discoveryrule.py:DiscoveryRules.test_delete_discovery_rule_1
.
----------------------------------------------------------------------
Ran 1 test in 132.546s

OK

# nosetests tests/foreman/ui/test_discoveryrule.py:DiscoveryRules.test_update_discovery_rule_1
.
----------------------------------------------------------------------
Ran 1 test in 28.837s

OK
```

### CODE REVIEW

File: tests/foreman/ui/test_discoveryrule.py [hunk 1]
```diff
@@ -94,16 +93,18 @@ def test_negative_create_discovery_rule_2(self, name):
 
         """
         with Session(self.browser) as session:
-            make_discoveryrule(session, name=name,
-                               hostgroup=self.host_group.name)
-            self.assertIsNotNone(self.discoveryrules.wait_until_element(
-                common_locators['name_haserror']
-            ))
-            self.assertIsNone(self.discoveryrules.search(name))
+            for name in "", "  ":
```
Comment (oshtaier) [inline tests/foreman/ui/test_discoveryrule.py]: for name in '', '  '

**Comment by oshtaier** [comment]
ACK pending non-blocking comment

File: tests/foreman/ui/test_discoveryrule.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/ui/test_discoveryrule.py]: I will combine this to the other negative test so I can use invalid_values_list() helper

Comment (oshtaier) [inline tests/foreman/ui/test_discoveryrule.py]: yeah, thank you. I thought to point on that, but decided to not ping you as there are a lot of PRs already :)

**Comment by sthirugn** [comment]
Comments addressed.
```sh
# nosetests tests/foreman/ui/test_discoveryrule.py
.........
----------------------------------------------------------------------
Ran 9 tests in 434.275s

OK
```

File: tests/foreman/ui/test_discoveryrule.py [hunk 2]
```diff
@@ -67,26 +67,7 @@ def test_positive_create_discovery_rule_2(self):
...
-            self.assertIsNotNone(self.discoveryrules.wait_until_element(
-                common_locators['name_haserror']
-            ))
-            self.assertIsNone(self.discoveryrules.search(name))
-
-    @data("", "  ")
-    @run_only_on('sat')
-    def test_negative_create_discovery_rule_2(self, name):
-        """@Test: Create Discovery Rule with blank and whitespace in name
+        """@Test: Create Discovery Rule with invalid names
```
Comment (sthirugn) [inline tests/foreman/ui/test_discoveryrule.py]: two negative tests are combined to one using invalid_values_list() helper

**Comment by oshtaier** [comment]
ACK

File: tests/foreman/ui/test_discoveryrule.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/ui/test_discoveryrule.py]: No problem :)

File: tests/foreman/ui/test_discoveryrule.py [hunk 3]
```diff
@@ -113,12 +96,15 @@ def test_negative_create_discovery_rule_3(self, limit):
...
+                    make_discoveryrule(session, name=name, host_limit=limit,
+                                       hostgroup=self.host_group.name)
+                    self.assertIsNotNone(
+                        self.discoveryrules.wait_until_element(
+                            common_locators['haserror'])
+                    )
+                    self.assertIsNone(self.discoveryrules.search(name))
 
     @run_only_on('sat')
     def test_negative_create_discovery_rule_4(self):
```
Comment (abalakh) [inline tests/foreman/ui/test_discoveryrule.py]: Rename to `test_negative_create_discovery_rule_3`

**Comment by abalakh** [comment]
ACK pending non-blocker comment and a rebase

File: tests/foreman/ui/test_discoveryrule.py (hunk 3 above)
Comment (sthirugn) [inline tests/foreman/ui/test_discoveryrule.py]: :+1:

**Comment by sthirugn** [comment]
Comments addressed and rebased.

**Comment by sthirugn** [comment]
part of https://github.com/SatelliteQE/robottelo/issues/2769

**Pull request merged**